### PR TITLE
Basic structure of CSV generation from output document batch processing

### PR DESCRIPTION
**Pull request by stevenwilkin**
Well gents, what do you think?

### CODE REVIEW

**Comment by stevenwilkin** [comment]
Updated this to reflect latest changes on `master`

File: app/domain/batch_process.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class BatchProcess
+  attr_reader :output_documents
+
+  def initialize(output_documents)
+    @output_documents = output_documents
+  end
+
+  def csv
+    [headers].concat(output_documents.map(&:csv)).join
```
Comment (andrewgarner) [inline app/domain/batch_process.rb]: I'm not quite sure about this. We are trusting (we know) that `output_document.csv` is implemented using `OutputDocument::CSVRenderer`.

If `output_document` returned a hash we could keep all knowledge of how to generate a CSV private to this class and also use the standard library to generate the CSV.

File: app/domain/batch_process.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  private
+
+  def headers
+    OutputDocument::CSVRenderer::FIELDS.join('|') + "\n"
```
Comment (andrewgarner) [inline app/domain/batch_process.rb]: This has potential to catch is out. We're trusting that fields won't need escaping can just be concatenated. Why not use http://ruby-doc.org/stdlib-2.2.1/libdoc/csv/rdoc/index.html ?

Comment (stevenwilkin) [inline app/domain/batch_process.rb]: I had thought the headers would need some sort of special treatment but now that you mention it I think treating them just like another row would be the thing to do per [CSVRenderer#render](https://github.com/guidance-guarantee-programme/output/blob/master/app/domain/output_document/csv_renderer.rb#L20-L22)

Comment (gjvis) [inline app/domain/batch_process.rb]: Perhaps move this to `CSVRenderer.header_row`

File: app/domain/batch_process.rb (hunk 1 above)
Comment (gjvis) [inline app/domain/batch_process.rb]: how about something like:
```ruby
def csv
  [headers].concat(output_document.map(&method(:csv_row))
end

def csv_row(output_document)
  OutputDocument::CSVRenderer.new(output_document).render
end
```

Comment (gjvis) [inline app/domain/batch_process.rb]: or move this entire class to be a class method on CSVRenderer:

```ruby
def csv(output_documents)
  csv = []
  csv << header_row
  output_documents.each do |doc|
    csv << new(doc).render
  end

  csv
end
``

Comment (gjvis) [inline app/domain/batch_process.rb]: or rename this file `CSVRenderer`, and rename the current `CSVRenderer` to `CSVRowRenderer`

Comment (andrewgarner) [inline app/domain/batch_process.rb]: With `header_row ` we still know that `output_documents` are rendered by `CSVRenderer`.

**Comment by stevenwilkin** [comment]
Per discussion, I'm going to close this PR and rework this class into a `CSVRenderer` which then makes use of `OutputDocument::CSVRowRenderer`

File: app/domain/batch_process.rb (hunk 1 above)
Comment (gjvis) [inline app/domain/batch_process.rb]: well yes, something has to, this is that thing.

**Comment by stevenwilkin** [comment]
The batch process will then be orchestrated by another object

**Pull request closed without merging**